internal/controller/stages/regular_stages_test.go

```diff
@@ -6577,3 +6578,104 @@ func Test_buildFreightSummary(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcile_ShardMatching(t *testing.T) {
```

Comment:
It would probably be both more succinct and more clear if a "shard doesn't match" test case were added to the existing `TestRegularStageReconciler_Reconcile()` tests. It would logically fall after the "ignores control flow stage" case.